librz/include/rz_basefind.h

```diff
@@ -17,11 +17,33 @@ extern "C" {
 #define RZ_BASEFIND_SCORE_MIN_VALUE   (1)
 
 typedef struct rz_basefind_t {
-	ut64 candidate;
-	ut32 score;
+	ut64 candidate; ///< Candidate physical base address
+	ut32 score; ///< Score of the candidate address
 } RzBaseFindScore;
 
-RZ_API RZ_OWN RzList *rz_basefind(RZ_NONNULL RzCore *core, ut32 pointer_size);
+typedef struct rz_basefind_info_t {
+	ut32 thread_idx; ///< Thread number
+	ut32 n_threads; ///< Total number of threads
+	ut64 begin_address; ///< Thread begin address
+	ut64 current_address; ///< Thread current address
+	ut64 end_address; ///< Thread end address
+	ut32 percentage; ///< Current percentage of the thread scan
+} RzBaseFindThreadInfo;
+
+typedef bool (*RzBaseFindThreadInfoCb)(const RzBaseFindThreadInfo *th_info, void *user);
```

Comment:
What is this used for exactly? Also, the doc for the fields might be preciser i guess?
I agree unit tests would help.

Reply:
i do not think is a good idea to implement unit tests for this, since this will require a big amount of time to run (compared to the other tests).